types/time.go

```diff
@@ -2225,9 +2203,21 @@ func GetFormatType(format string) (isDuration, isDate bool) {
 			isDuration, isDate = false, false
 			break
 		}
-		if _, ok := durationTokens[token]; ok {
+		var durationTokens bool
+		var dateTokens bool
+		if len(token) >= 2 {
```

Comment:
if len(token) == 2 &&  token[0] = '%'

Reply:
Thanks, not it will fater.
At beginning, I have thought it will make it slow, but I'm wrong.
```
xiekeyi@ubuntu:~$ go test -bench=. -benchmem -benchtime=5s -count=3 a_test.go 
goos: linux
goarch: amd64
BenchmarkSwitch1-4   	100000000	        50.1 ns/op	       0 B/op	       0 allocs/op
BenchmarkSwitch1-4   	100000000	        59.2 ns/op	       0 B/op	       0 allocs/op
BenchmarkSwitch1-4   	100000000	        58.4 ns/op	       0 B/op	       0 allocs/op
BenchmarkSwitch2-4   	100000000	        71.3 ns/op	       0 B/op	       0 allocs/op
BenchmarkSwitch2-4   	100000000	        74.8 ns/op	       0 B/op	       0 allocs/op
BenchmarkSwitch2-4   	100000000	        71.2 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	command-line-arguments	38.961s
xiekeyi@ubuntu:~$ cat a_test.go 
package types

import (
	"math/rand"
	"testing"
)

func testSwitch1(format string) (a, b bool) {
	a, b = false, false
	var ok bool
	if len(format) >= 2 && format[0] != '%' {
		switch format[1] {
		case 'h', 'H', 'i', 'I', 's', 'S', 'k', 'l':
			ok = true
		}
	}

	if ok {
		a, b = true, true
	}
	return a, b

}

func testSwitch2(format string) (a, b bool) {
	a, b = false, false
	var ok bool
	if len(format) >= 2 {
		switch format[1] {
		case 'h', 'H', 'i', 'I', 's', 'S', 'k', 'l':
			ok = true
		}
	}

	if ok {
		a, b = true, true
	}
	return a, b

}

func BenchmarkSwitch1(b *testing.B) {
	s := []string{"%h", "%H", "%i", "%a", "%s", "%S"}
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		_, _ = testSwitch1(s[rand.Intn(6)])
	}
}

func BenchmarkSwitch2(b *testing.B) {
	s := []string{"%h", "%H", "%i", "%a", "%s", "%S"}
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		_, _ = testSwitch2(s[rand.Intn(6)])
	}
}
```